### ISSUE TITLE
v4 Stack.map indexes reverse of v3?

### ISSUE THREAD

**Issue by chrisnojima** [facebook/immutable-js#1375]
Hey,

Noticed this difference and didn't see it mentioned in the v4.0.0-rc5 release notes. Expected?

v3
```
> I.Stack(['one', 'two', 'three']).map((val, idx) => console.log(`val: ${val} idx: ${idx}`))
val: three idx: 0
val: two idx: 1
val: one idx: 2
Stack [ undefined, undefined, undefined ]
```

v4.0.0-rc7
```
> I.Stack(['one', 'two', 'three']).map((val, idx) => console.log(`val: ${val} idx: ${idx}`))
val: three idx: 2
val: two idx: 1
val: one idx: 0
Stack [ undefined, undefined, undefined ]
```

Thanks

**Comment by leebyron** [comment]
I believe this was a bug fix - let me do a quick investigation

**Comment by leebyron** [comment]
Yep, this was a bug fix that landed in v4.0.0-rc.1 - my apologies for not making that list of bug fixes explicit:

For example:

v3.8.1:

```js
> s = Stack([1,2,3])
Stack [ 1, 2, 3 ]
> s.forEach(console.log)
1 0 Stack [ 1, 2, 3 ]
2 1 Stack [ 1, 2, 3 ]
3 2 Stack [ 1, 2, 3 ]
3
> s.map(console.log)
3 0 Stack [ 3, 2, 1 ]
2 1 Stack [ 3, 2, 1 ]
1 2 Stack [ 3, 2, 1 ]
```

v4.0.0-rc.1:


```js
> s = Stack([1,2,3])
Stack [ 1, 2, 3 ]
> s.forEach(console.log)
1 0 Stack [ 1, 2, 3 ]
2 1 Stack [ 1, 2, 3 ]
3 2 Stack [ 1, 2, 3 ]
3
> s.map(console.log)
3 2 Stack [ 3, 2, 1 ]
2 1 Stack [ 3, 2, 1 ]
1 0 Stack [ 3, 2, 1 ]
```

Mapping works through the values in reverse for a Stack since they're computed during insertion into the new Stack, which is done in reverse to maintain the same order as the original. The bug fix was to ensure the indexes properly reflected their position in the Stack, rather than the (reversed) order of calling the map function.
